sonic-net/sonic-buildimage
# [vs]: BMv2-based DPU KVM image

#### Why I did it
1.	Provide POC for development and cooperation.
2.	Test for CI (Azp)

##### Work item tracking
- Microsoft ADO **(number only)**: 27515178

#### How I did it

HLD: https://github.com/sonic-net/SONiC/pull/1634

#### How to verify it

1. Download the vsdpu image from Azp
2. check the syncd links to the correct DASH SAI
```
admin@vlab-01:~$ docker exec -ti syncd bash
root@vlab-01:/# dpkg -l | grep sai
ii  libsai                                 1.0.0                          amd64        This package contains DASH libsai
root@vlab-01:/# dpkg -L libsai
/.
/etc
/etc/dash
/etc/dash/dash_pipeline.json
/etc/dash/dash_pipeline_ir.json
/etc/dash/dash_pipeline_p4rt.json
/etc/dash/dash_pipeline_p4rt.txt
/usr
/usr/lib
/usr/lib/x86_64-linux-gnu
/usr/lib/x86_64-linux-gnu/libsai.so
root@vlab-01:/# which syncd
/usr/bin/syncd
root@vlab-01:/# ldd /usr/bin/syncd | grep sai
        libsaimetadata.so.0 => /usr/lib/x86_64-linux-gnu/libsaimetadata.so.0 (0x00007f949e99b000)
        libsaimeta.so.0 => /usr/lib/x86_64-linux-gnu/libsaimeta.so.0 (0x00007f949e8ca000)
        libsai.so => /usr/lib/x86_64-linux-gnu/libsai.so (0x00007f949e417000)
root@vlab-01:/# md5sum /usr/lib/x86_64-linux-gnu/libsai.so
9e86fa2f49a824837064d1b8f95b7e48  /usr/lib/x86_64-linux-gnu/libsai.so
root@vlab-01:/# ls /etc/dash
dash_pipeline.json  dash_pipeline_ir.json  dash_pipeline_p4rt.json  dash_pipeline_p4rt.txt
```

3. Check the dash engine logs(BMv2) likes following
```
admin@vlab-01:~$ docker logs dash_engine
Calling target program-options parser
Adding interface eth1 as port 0
[11:20:12.458] [bmv2] [D] [thread 7] Adding interface eth1 as port 0
[11:20:12.500] [bmv2] [E] [thread 7] Add port operation failed
Adding interface eth2 as port 1
[11:20:12.500] [bmv2] [D] [thread 7] Adding interface eth2 as port 1
[11:20:12.556] [bmv2] [E] [thread 7] Add port operation failed
Server listening on 0.0.0.0:9559
[11:22:09.706] [bmv2] [D] [thread 23] Set default default entry for table 'tbl_dash_pipeline241': dash_pipeline241 -
[11:22:09.706] [bmv2] [D] [thread 23] Set default default entry for table 'dash_ingress.vip': dash_ingress.deny -
[11:22:09.706] [bmv2] [D] [thread 23] Set default default entry for table 'tbl_dash_pipeline247': dash_pipeline247 -
[11:22:09.706] [bmv2] [D] [thread 23] Set default default entry for table 'dash_ingress.direction_lookup_stage.direction_lookup': dash_ingress.direction_lookup_stage.set_inbound_direction -
[11:22:09.706] [bmv2] [D] [thread 23] Set default default entry for table 'dash_ingress.appliance': NoAction -
[11:22:09.706] [bmv2] [D] [thread 23] Set default default entry for table 'tbl_eni_lookup33': eni_lookup33 -
[11:22:09.706] [bmv2] [D] [thread 23] Set default default entry for table 'tbl_eni_lookup33_0': eni_lookup33_0 -
[11:22:09.706] [bmv2] [D] [thread 23] Set default default entry for table 'tbl_eni_lookup33_1': eni_lookup33_1 -
[11:22:09.706] [bmv2] [D] [thread 23] Set default default entry for table 
...
[11:23:32.035] [bmv2] [D] [thread 23] Set default default entry for table 'tbl_metering_update92': metering_update92 -
[11:23:32.035] [bmv2] [D] [thread 23] Set default default entry for table 'tbl_metering_update94': metering_update94 -
[11:23:32.035] [bmv2] [D] [thread 23] Set default default entry for table 'tbl_metering_update97': metering_update97 -
[11:23:32.035] [bmv2] [D] [thread 23] Set default default entry for table 'dash_ingress.metering_update_stage.meter_bucket': NoAction -
[11:23:32.035] [bmv2] [D] [thread 23] Set default default entry for table 'tbl_metering_update103': metering_update103 -
[11:23:32.035] [bmv2] [D] [thread 23] Set default default entry for table 'tbl_metering_update105': metering_update105 -
[11:23:32.035] [bmv2] [D] [thread 23] Set default default entry for table 'dash_ingress.metering_update_stage.eni_meter': NoAction -
[11:23:32.035] [bmv2] [D] [thread 23] Set default default entry for table 'tbl_drop_action': dash_ingress.drop_action -
[11:23:32.049] [bmv2] [D] [thread 23] simple_switch target has been notified of a config swap
```



#### Which release branch to backport (provide reason below if selected)




#### Tested branch (Please provide the tested image version)




#### Description for the changelog




#### Link to config_db schema for YANG module changes


#### A picture of a cute animal (not mandatory but encouraged)
🐯
